File: WebSiteRoute.Application/FlightAltitudes/GetFlightAltitudeDashboardQuery.cs
using WebSiteRoute.Application.Abstractions;
using WebSiteRoute.Domain.Services;

namespace WebSiteRoute.Application.FlightAltitudes;

public sealed class GetFlightAltitudeDashboardQuery(
    IAltitudeReadingReader altitudeReadingReader,
    IFlightStatusMessageProvider flightStatusMessageProvider)
{
    public async Task<FlightAltitudeDashboard> ExecuteAsync(CancellationToken cancellationToken = default)
    {
        var readings = await altitudeReadingReader.GetReadingsAsync(cancellationToken);
        var sampleAltitude = readings.FirstOrDefault()?.Altitude ?? 0;
        var message = flightStatusMessageProvider.GetStatusMessage(sampleAltitude);

        return new FlightAltitudeDashboard(readings, message, sampleAltitude);
    }
}

File: src/main/java/com/websiteroute/application/GetFlightAltitudeDashboardQuery.java
package com.websiteroute.application;

import com.websiteroute.domain.FlightStatusMessageProvider;
import org.springframework.stereotype.Service;

@Service
public class GetFlightAltitudeDashboardQuery {
    private final AltitudeReadingRepository repository; private final FlightStatusMessageProvider messages;
    public GetFlightAltitudeDashboardQuery(AltitudeReadingRepository repository, FlightStatusMessageProvider messages) { this.repository = repository; this.messages = messages; }
    public FlightAltitudeDashboard execute() {
        var readings = repository.getReadings(); int altitude = readings.stream().findFirst().map(r -> r.altitude()).orElse(0);
        return new FlightAltitudeDashboard(readings, messages.getStatusMessage(altitude), altitude);
    }
}
